### PR TITLE
EES-2359 Add data guidance text file to 'All files' zip

### PR DESCRIPTION
**Pull request by ntsim**
This PR adds a new `data-guidance.txt` file to the 'All files' zip. This is essentially just a text version of the metadata guidance file (this will be renamed to data guidance across the frontend in [EES-2235](https://dfedigital.atlassian.net/browse/EES-2235)).

To generate this text file, we leverage the HTML to text converter created in EES-2484 (PR #2644).

## Other changes

- Fixed `blockquote` elements in converted text having missing trailing lines.
- Refactored `Startup` files to use a common `StartupUtils.AddPersistenceHelper` util to setup their persistence helpers.

### CODE REVIEW

**Comment by markhiveit** [comment]
I'm getting some unit test failures with this error message. Be interested to see whether other windows users are getting these:
image

**Review by markhiveit — CHANGES_REQUESTED** [review]

**Comment by markhiveit** [comment]
> I'm getting some unit test failures with this error message. Be interested to see whether other windows users are getting these:
> image

This was due to not being on the latest version of the branch - I checked out EES-2359 rather than ees-2359, and it seems a casesensitivity issue somewhere meant that I could checkout EES-2359, but when pulling from EES-2359 it wouldn't pulled the latest from ees-2359.

File: src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/Interfaces/IReleaseService.cs [hunk 1]
```diff
@@ -10,17 +11,19 @@ namespace GovUk.Education.ExploreEducationStatistics.Publisher.Services.Interfac
...
+        Task<IEnumerable<Release>> List(IEnumerable<Guid> ids);
 
         Task<IEnumerable<Release>> GetAmendedReleases(IEnumerable<Guid> releaseIds);
 
         Task<List<FileInfo>> GetDownloadFiles(Release release);
 
         Task<List<File>> GetFiles(Guid releaseId, params FileType[] types);
 
-        Task<Release> GetLatestRelease(Guid publicationId, IEnumerable<Guid> includedReleaseIds);
+        Task<Release?> GetLatestRelease(Guid publicationId, IEnumerable<Guid> includedReleaseIds);
```
[the remark this answers is not in the document]
Comment (ntsim) [reply src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/Interfaces/IReleaseService.cs:26]: Good point, I was mostly just looking at the return type in isolation. I'll change it to non-nullable using `Last`.

File: src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ReleaseService.cs [hunk 2]
```diff
@@ -46,15 +47,27 @@ public class ReleaseService : IReleaseService
...
             return await _contentDbContext.Releases
                 .Include(release => release.Publication)
                 .Include(r => r.PreviousVersion)
                 .SingleOrDefaultAsync(release => release.Id == id);
         }
 
-        public async Task<IEnumerable<Release>> GetAsync(IEnumerable<Guid> ids)
+        public async Task<Release> Get(Guid id)
+        {
+            var release = await Find(id);
```
[the remark this answers is not in the document]
Comment (ntsim) [reply src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ReleaseService.cs:60]: Think it's probably a better idea to have both I'd say. It's reasonable that there may be a future use-case for it being null, and there isn't any particular harm in having it.

**Comment by ntsim** [comment]
Do not merge - need to confirm if additional modifications to the 'All files' zip should be made.

**Review by markhiveit — APPROVED** [review]

**Pull request merged**